src/client/options/mod.rs

```diff
@@ -1372,11 +1408,22 @@ impl ClientOptionsParser {
                 self.zlib_compression = Some(i);
             }
 
-            _ => {
-                return Err(ErrorKind::ArgumentError {
-                    message: "invalid option warning".to_string(),
+            other => {
+                let (jaro_winkler, option) = URI_OPTIONS.iter().fold((0.0, ""), |acc, option| {
+                    let jaro_winkler = jaro_winkler(option, other).abs();
+                    if jaro_winkler > acc.0 {
+                        return (jaro_winkler, option);
+                    }
+                    acc
+                });
+                let mut message = format!("{} is an invalid option", other);
+                if jaro_winkler >= 0.84 {
```

Comment:
I'm curious about how the value 0.84 was picked; was this just trial and error?

Reply:
Yep, mostly trial and error based on the strings in the test and some other ones I tried out, plus looking through some other examples of Jaro-Winkler being used.